presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSession.java

```diff
@@ -58,4 +58,9 @@ default boolean isReadConstraints()
     {
         return false;
     }
+
+    default void addWarningMessage(PrestoWarning warning)
+    {
+        return;
+    }
```

Comment:
Just `getWarningCollector`. This will not be a default interface as the collector is always available.

Reply:
Change to getWarningCollector to return warningCollector.